UN-OCHA/unocha-site
# [UNO-682] Regional donors data

Refs: UNO-682, OHA-55

This PR replaces the fields on the donors paragraph with a "KeyFigure" field (donors for a country *or* full list of ocha donors) and a "KeyFigurePresence" field (donors for a country office or regional office).

The templates etc. are unchanged. 

### Editorial notes

There 2 fields to select donors data, one to select data for a country and one to select data for a regional or country office.

Editors should use the country one for responses and the other one for regions but not both at the same time (that would obviously display 2 sets of donors data).

To show the full OCHA donor list like on https://unocha.org/funding, select "Global" as country then year and "Top donors".

### Follow-ups

**Fields**

The previous fields (donor data type and country) are still there but hidden. They should be removed but there is probably existing data on dev or local that may be problematic. So after this PR is merged and deployed, the following should be done:

1. On local, dev and preview -> go to `/admin/structure/paragraphs_type/donors/fields` and delete the `type` and `countries` fields.
2. On local, export the config with the deleted fields and create a PR
3. Merge and deploy to dev and preview

**Patch**

There is a patch equivalent to https://github.com/UN-OCHA/ocha_key_figures/pull/8 for the `ocha_key_figures` module. After that gets merged then we should remove the patch and bump the version of the module in composer.json.

### Tests

1. Checkout the branch, clear the cache, import the config
2. Edit a response, 
3. Add a "donors" paragraph and select some donors data from CBPF, FTS or OCT using the "country" field
4. Check that the data appears properly
5. Edit a region
6. Repeat (3) and (4) but use the "country/regional office" field
7. Edit or create the "/funding" page
8. Repeat (3) and (4) but select "Global" as country and "Top donors" as figure
